tests/test_validation.py

```diff
@@ -1083,3 +1054,901 @@ def test_login(self, dmge: DataModelGraphExplorer) -> None:
+
+        with patch.object(
+            schematic.models.validate_attribute.ValidateAttribute,
+            "_get_target_manifest_dataframes",
+            return_value={"syn1": cross_val_df1},
+        ):
+            assert va_obj.cross_validation(val_rule, Series([])) == ([], [])
+            assert va_obj.cross_validation(val_rule, Series(["A"])) == ([], [])
+            assert va_obj.cross_validation(val_rule, Series(["A", "A"])) == ([], [])
+            assert va_obj.cross_validation(val_rule, Series(["A", "B"])) == ([], [])
+            assert va_obj.cross_validation(val_rule, Series(["A", "B", "C"])) == (
+                [],
+                [],
+            )
+            assert va_obj.cross_validation(val_rule, Series(["A", "B", "C", "C"])) == (
+                [],
+                [],
+            )
+
+    def test_cross_validation_value_match_atleast_one_rules_errors(
+        self, va_obj: ValidateAttribute, cross_val_df1: DataFrame
+    ):
+        """Tests for cross manifest validation for matchAtLeastOne value rules"""
+        val_rule = "matchAtLeastOne Patient.PatientID value error"
+
+        with patch.object(
+            schematic.models.validate_attribute.ValidateAttribute,
+            "_get_target_manifest_dataframes",
+            return_value={"syn1": cross_val_df1},
+        ):
+            errors, _ = va_obj.cross_validation(
+                val_rule, Series(["D"], index=[0], name="PatientID")
+            )
+            assert len(errors) == 1
+
+    def test_cross_validation_match_exactly_ne_value_rules_passing(
+        self, va_obj: ValidateAttribute, cross_val_df1: DataFrame
+    ):
+        """Tests for cross manifest validation for matchExactlyOne value rules"""
+        val_rule = "matchExactlyOne Patient.PatientID value error"
+
+        with patch.object(
+            schematic.models.validate_attribute.ValidateAttribute,
+            "_get_target_manifest_dataframes",
+            return_value={"syn1": cross_val_df1},
+        ):
+            assert va_obj.cross_validation(val_rule, Series([])) == ([], [])
+            assert va_obj.cross_validation(val_rule, Series(["A"])) == ([], [])
+            assert va_obj.cross_validation(val_rule, Series(["A", "A"])) == ([], [])
+            assert va_obj.cross_validation(val_rule, Series(["A", "B"])) == ([], [])
+            assert va_obj.cross_validation(val_rule, Series(["A", "B", "C"])) == (
+                [],
+                [],
+            )
+            assert va_obj.cross_validation(val_rule, Series(["A", "B", "C", "C"])) == (
+                [],
+                [],
+            )
+
+        with patch.object(
+            schematic.models.validate_attribute.ValidateAttribute,
+            "_get_target_manifest_dataframes",
+            return_value={"syn1": cross_val_df1, "syn2": cross_val_df1},
+        ):
+            assert va_obj.cross_validation(val_rule, Series([])) == ([], [])
```

Comment:
I am confused why this case too there's no error returns when an empty series is provided.. if rule is match exactly one, shouldn't this raise an error?

Reply:
Actually I think you're right, let me get back to you on this one.

Reply:
OK this does NOT work how I thought. For `value` the code goes through and combines all the target manifests together into one and then does the comparison with the manifest being validated.

Then it does this:

```
        # Find values that are present in the source manifest,
        # but not present in the target manifest
        missing_values = manifest_col[~manifest_col.isin(concatenated_target_column)]

        # Find values that duplicated in the concatenated target column,
        # and also present in the source manifest column
        duplicated_values = manifest_col[
            manifest_col.isin(
                concatenated_target_column[concatenated_target_column.duplicated()]
            )
        ]

        # Find values that are repeated between the manifest column and concatenated target column
        repeat_values = manifest_col[manifest_col.isin(concatenated_target_column)]
```
When the column being validated is empty `missing_values` `duplicated_values` adn `repeat_values` are all empty.
These values get sent to another method to determine if there is an error:

```
        elif "matchExactlyOne" in val_rule and (
            validation_output.duplicated_values.any()
            or validation_output.missing_values.any()
        ):
            invalid_rows, invalid_entry = self._merge_format_invalid_rows_values(
                validation_output.duplicated_values, validation_output.missing_values
            )
```
so for matchExactlyOne to have an error(or warning) there have to be duplicated_values and missing_values, which there aren't if the column being validated has no values in it